### DISCUSSION TITLE
Discontinued? [Nope not anymore]

### DISCUSSION THREAD

**Discussion by Sixsonic** [AlexAsplund/Vanchor#25] [General]
Hi, i found this project, and im willing to give it a shot. Is it working or not. The crucial part i need is vanchor, would love to have an anchor function with trolling motor

**Comment by Kenaiwolf** [comment]
+1

**Reply by krautech** [reply]
Yeah I'd love to see this project continued..happy to help in any way

**Reply by CobraXLazer** [reply]
Me to!

**Comment by AlexAsplund** [comment]
@Sixsonic @krautech @CobraXLazer @Kenaiwolf 

Started from scratch with claude. Currently everything is through a boat physics simulator but should be easy enough hooking into the real thing.

See the repo - it now have most of the features I thought about during the years.

I'm also CAD'ing a new servo in onshape. But that's not really my strong area so we'll see.

**Reply by Sixsonic** [reply]
Amazing keep me updated!!!!!

**Reply by jkknibbe99** [reply]
@AlexAsplund, I Would love to help out with this project. I've been working on a custom control for an old Minn Kota Power Drive myself. I think a lot of this tech would crossover as my motor already has electric steering. I have a decent amount of Onshape experience as well 👍

**Reply by AlexAsplund** [reply]
@jkknibbe99 - Sounds great! I added you as a collaborator if you want to try things out.

Linking the first iteration of the new servo. It uses a 20rpm wormgear motor + an AS5600 for feedback. Was planning on having the arduino + driver in another box to begin with. Mind that i haven't tried printing it yet - need to unclog my prusa first 😄 

https://cad.onshape.com/documents/0cd9f6cea03bdb2bf6d750e6/w/0cfcf429ac87db948a953e94/e/b4b8c630f36b096ae46ff4ce?renderMode=0&uiState=6a451955398f55545b02b00a

**Reply by AlexAsplund** [reply]
Oh btw, I added some device related stuff today so that you can add custom "drivers" for steering, compass etc. Should make it easy enough so that you don't need custom hardware that uses the "vanchor protocol".

It adds actions and config for the devices as well. For example calibration, declination settings etc for my HWT901b.

<img width="421" height="1181" alt="bild" src="https://github.com/user-attachments/assets/31d7dc87-d787-4135-95e3-cfcea5638b1c" />

**Reply by jkknibbe99** [reply]
The custom drivers idea would be huge! This would allow the software for use independent of your hardware. Maybe I will focus on finalizing my personal Minn Kota project hardware and then see how I can integrate it with Vanchor 👍

The CAD looks like a good start. Let me know if I can help out on that side of things.

**Reply by AlexAsplund** [reply]
Agreed! Would be cool for example to have a special device type for minn kota motors.

Currently printing :)

Also, I had an old PCB design that I never finished. But i had claude remake it:
<img width="1568" height="872" alt="bild" src="https://github.com/user-attachments/assets/3bd0ade5-211a-4fdf-ac92-523b0d8afedf" />

Basically you can connect a RPI/Orange Zero to it to run Vanchor it has a lot of stuff built in, like:
- A Raspberry Pico that controls the on-board servo controller for the steering as well as sending PWM etc to a thruster driver.
-  2 UART interfaces for stuff like the HWT901
- Prepared for NMEA2000 / CAN interface
- On board pico is reprogrammable by the Orange/RPI

I think being able to interface with NMEA2000 would be really cool. Then we could take in data from other n2k devices (AIS, Main Engine Status/RPM etc) and also send commands. There is for example an N2K thruster command that could be used to control the trolling engine.

I'm going to inspect the tracing and connections, but it seems fine.

There is also a seperate thruster driver with PWM in and prepared for N2K/CAN that should handle 12V/100A or 24V/50A.

Total cost for ordering PCB's + parts is ~$150 (That includes both the helm controller and thruster driver).

https://github.com/AlexAsplund/vanchor-pcb

**Reply by AlexAsplund** [reply]
@jkknibbe99 

The driver functionality is implemented.

I added drivers for my ublox M9N gps and HWT901B imu/compass and it's working well.

For the motor there are 2 modes: The "regular" mode that assumes they the same controller controls both steering + motor and an "advanced" mode where you can have thrust driver and steering servo on different ports.

There's experimental and untested support for nmea 2000 as well if you want to go that route 😁
